src/components/nutrition/label/nutrition-popover.tsx

```diff
@@ -0,0 +1,108 @@
+import {atom, useAtomValue} from "jotai";
+import {PopupState, bindPopper} from "material-ui-popup-state/hooks";
+import {useQuery} from "@tanstack/react-query";
+import React from "react";
+import {Box, ClickAwayListener, Popper, Tooltip, Typography} from "@mui/material";
+import {Placement} from "@floating-ui/utils";
+import {WarningOutlined} from "@mui/icons-material";
+
+import {buildGetFoodQuery} from "@/api/nutrition/foods";
+import NutritionLabel from "@/components/nutrition/label/nutrition-label";
+import {FoodLogEntry, NutritionMacroGoals} from "@/api/nutrition";
+import {formatServing} from "@/utils/food-amounts";
+
+export interface NutritionPopoverContext {
+  foodLog: FoodLogEntry | null;
+  foodId: number;
+}
+
+const EMPTY_CONTEXT: NutritionPopoverContext = {
+  foodLog: null,
+  foodId: 0,
+};
+
+export const nutritionPopoverFoodAtom = atom<NutritionPopoverContext>(EMPTY_CONTEXT);
+
+const NutritionPopover = function ({macroGoals, popupState, placement, offset}: {
+  macroGoals: NutritionMacroGoals,
+  offset?: [number, number],
+  popupState: PopupState,
+  placement?: Placement
+}) {
+  const context = useAtomValue(nutritionPopoverFoodAtom);
+  const {data: food} = useQuery({
+    ...buildGetFoodQuery(context.foodId),
+    enabled: context.foodId > 0
+  });
+
+  return (
+    <Box>
+      {(food && food.defaultUnit && food.defaultServingSize) && (
+        <Popper
+          {...bindPopper(popupState)}
+          placement={placement || "left"}
+          anchorEl={popupState.anchorEl}
+          modifiers={[
+            {
+              name: "arrow",
+              enabled: true,
+              options: {
+                element: popupState.anchorEl,
+              },
+            },
+            {
+              name: "offset",
+              enabled: true,
+              options: {
+                offset: offset || [0, 0],
+              },
+            },
+          ]}
+        >
+          <ClickAwayListener onClickAway={popupState.close}>
+            <Box className="bg-slate-50 dark:bg-slate-900 p-21">
```

Comment:
Overlays like popovers should use something like `<Paper>` to provide a border and shadow, otherwise it's harder to distinguish from the content below.

Also: `p-21` is not a valid tailwind class (looks like it goes up to 12) so this doesn't do anything

Reply:
I had a Paper at first but the popper wouldn't close. It seemed to work with a Box instead. Maybe it didn't close for another reason. I'll try Paper again.